tests/manager_arg_passer.c

```diff
@@ -72,36 +23,40 @@ prep_filename(char* filename)
     return filename;
 }
 
-void
-args_clean(void** comp_args, size_t arg_count)
-{
-    for (size_t i = 0; i < arg_count; ++i)
-    {
-        free(comp_args[i]);
-    }
-    free(comp_args);
-}
-
 int
 main(int argc, char** argv)
 {
     // Initial setup
     manager_ddc = cheri_ddc_get();
     setup_intercepts();
 
-    assert(argc >= 3 && "Expect at least two arguments: binary file for compartment, and at least one argument to pass to compartment.");
+    assert(argc >= 4 && "Expect at least three arguments: binary file for compartment, entry function for compartment, and at least one argument to pass to compartment function.");
     char* file = prep_filename(argv[1]);
+    char* file_config = malloc(sizeof(file) + sizeof(comp_config_suffix) + 1);
+    strcpy(file_config, file);
+    strcat(file_config, comp_config_suffix);
+    FILE* comp_config_fd = fopen(file_config, "r");
+    free(file_config);
+    struct ConfigEntryPoint* cep = NULL;
+    size_t entry_point_count = 0;
+    char** entry_points = NULL;
+    if (comp_config_fd)
+    {
+        cep = parse_compartment_config(comp_config_fd, &entry_point_count);
+        fclose(comp_config_fd);
+    }
 
-    struct CompEntryPoints* cep = get_entry_points(file);
-    struct Compartment* arg_comp = comp_from_elf(file, cep->entry_points);
+    struct Compartment* arg_comp = comp_from_elf(file, cep, entry_point_count);
     loaded_comp = arg_comp; // TODO
     log_new_comp(arg_comp);
     comp_map(arg_comp);
 
-
-    void** comp_args = get_entry_point_args(cep);
-    int comp_result = comp_exec(arg_comp, get_entry_point_fn(cep), comp_args, cep->arg_count);
+    char* entry_func = argv[2];
```

Comment:
Probably should `assert` there are enough entries in `argv`.

Reply:
This is done further up [1].

[1] https://github.com/capablevms/CHERI-ELF-comp/pull/7/files/8e66303ac0eb7d30d0174792d0f1c62ad98ea007#diff-6e0cbcab5be8419f8d8accdd49cf5e8096fefb3f0f6f3105b8cb95662ba16347R33